src/bp/ui-admin/src/Pages/Account/ChangePassword.tsx

```diff
@@ -0,0 +1,128 @@
+import { Button, FormGroup, InputGroup, Intent } from '@blueprintjs/core'
+import React, { Component, Fragment } from 'react'
+import { RouteComponentProps } from 'react-router'
+import { Redirect } from 'react-router-dom'
+import { Alert, Card, CardBody, CardText, CardTitle } from 'reactstrap'
+
+import logo from '../../media/nobg_white.png'
+
+type Props = {
+  auth: any
+} & RouteComponentProps
+
+interface State {
+  loginUrl: string
+  email: string
+  password: string
+  newPassword: string
+  confirmPassword: string
+  error?: string
+}
+
+export default class ChangePassword extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      ...props.location.state,
+      newPassword: '',
+      confirmPassword: '',
+      error: undefined
+    }
+  }
+
+  updatePassword = async () => {
+    const { email, password, newPassword, confirmPassword, loginUrl } = this.state
+
+    if (newPassword !== confirmPassword) {
+      return this.setState({ error: `Passwords don't match` })
+    }
+
+    this.setState({ error: undefined })
+
+    try {
+      await this.props.auth.login({ email, password, newPassword }, loginUrl)
+    } catch (err) {
+      this.setState({ error: err.message })
+    }
+  }
+
+  handleNewPasswordChanged = e => this.setState({ newPassword: e.target.value })
+  handleConfirmChanged = e => this.setState({ confirmPassword: e.target.value })
+  handleKeyPressed = e => e.key === 'Enter' && this.updatePassword()
+
+  getReasonText = () => {
+    if (this.state.password === '') {
+      return 'This is the first time you run Botpress. Please pick a password.'
```

Comment:
is this possible ? someone with the first time running botpress should land on the register page ?

Reply:
It's only to customize the message displayed to the user. If the password is set, then it's an existing user. Otherwise, it's the first account